lineage/states/StateDistribution.py

```diff
@@ -36,12 +37,19 @@ def pdf(self, tuple_of_obs):  # user has to define how to calculate the likeliho
         # distribution observations), so the likelihood of observing the multivariate observation is just the product of
         # the individual observation likelihoods.
 
-        bern_ll = bern_pdf(tuple_of_obs[0], self.bern_p)
-        gamma_ll = gamma_pdf(tuple_of_obs[1], self.gamma_a, self.gamma_scale)
+        try:
+            bern_ll = bern_pdf(tuple_of_obs[0], self.bern_p)
+        except ZeroDivisionError:
+            assert False, f"{tuple_of_obs[0]}, {self.bern_p}"
+            
+        try:
+            gamma_ll = gamma_pdf(tuple_of_obs[1], self.gamma_a, self.gamma_scale)
+        except ZeroDivisionError:
+            assert False, f"{tuple_of_obs[1]}, {self.gamma_a}, {self.gamma_scale}"
```

Comment:
@aarmey I want this to be caught in `Analyze` where we have it set up to run our model multiple times in case it catches an AssertionError. However, the error we catch here is a ZeroDivisionError which I "convert" to a AssertionError so that the model reruns. This is because on **really rare** cases (like the first iteration of the model is just really bad and leads to bad parameter estimates which leads to a pdf evaluating to 0), instead of rerunning, the code throws a ZeroDivisionError which doesn't cause the model to rerun even though we want it to. Is this the way to do it? I'm referring to this line: https://github.com/meyer-lab/lineage-growth/blob/869d16bde9ad866e17669c014c4cccf414ad5a38/lineage/Analyze.py#L29

Reply:
Couldn't you catch the `ZeroDivisionError` in Analyze?

Reply:
Would it just be:

`except AssertionError or ZeroDivisionError:`

Reply:
Never mind, https://stackoverflow.com/questions/6470428/catch-multiple-exceptions-in-one-line-except-block I see how to do it.

Reply:
Fixed now.